### ISSUE TITLE
Sophia join does not work

### ISSUE THREAD

**Issue by alyapunov** [tarantool/tarantool#1239]
on branch master (1.6.7)
running one instance (master) with script
```
box.cfg{listen = 33013}
box.schema.create_space('test', {engine = 'sophia'})
box.space.test:create_index('test')
box.schema.user.grant('guest', 'read, write, execute', 'universe')
box.snapshot()
box.space.test:insert{1, 2, 3}
```
and running another instance (replica) with script
```
box.cfg{replication_source = 'localhost:33013'}
```
I got an error on replica
```
tarantool> 2015-12-30 18:20:30.772 [2539] main/105/applier/localhost:33013 I> can't read row
2015-12-30 18:20:30.772 [2539] main/105/applier/localhost:33013 sophia_index.cc:613 E> ER_TUPLE_FOUND: Duplicate key exists in unique index 'test' in space 'test'
```
note that there is no problem if the engine in space is memtx

**Comment by rtsisyk** [comment]
We reverted commit in master, but it didn't work for 1.6+.
https://github.com/tarantool/tarantool/commits/gh-1239-fix-join-1.6-tmp

**Comment by rtsisyk** [comment]
Fixed in Sophia today.
